server/src/main/java/com/thoughtworks/go/domain/JobAgentMetadata.java

```diff
@@ -77,10 +77,11 @@ public ClusterProfile clusterProfile() {
         return new ClusterProfile(id, pluginId, configProperties);
     }
 
-    private static String jsonizeElasticAgentProfile(ElasticProfile elasticProfile) {
+    private static String jsonizeElasticAgentProfile(ElasticProfile elasticProfile, String pluginId) {
         Map<String, Object> map = new LinkedHashMap<>();
         map.put("clusterProfileId", elasticProfile.getClusterProfileId());
-        map.put("pluginId", elasticProfile.getPluginId());
+        //todo: elastic agent extension v6 should remove plugin id.
+        map.put("pluginId", pluginId);
```

Comment:
\cc: @maheshp @bdpiparva 

As part of elastic agent extension v5 migrate-config call, we are passing along `elastic agent profile pluginId` to the plugin. Ideally, this shouldn't have been sent.

I am keeping this as is and adding a todo item for removing it in elastic agent extension v6.